src/nvim/editor/arbmark.c

```diff
@@ -0,0 +1,156 @@
+/*
+ * Same function but names are not limited to one char
+ * There is also no ex_command, just viml functions
+ */
+
+#include "nvim/vim.h"
+#include "nvim/globals.h"
+#include "nvim/mark.h"
+#include "nvim/memory.h"
+#include "nvim/map.h"
+#include "nvim/lib/kvec.h"
+#include "nvim/editor/arbmark.h"
```

Comment:
@timeyyy “API only” means that you need public interface in src/nvim/api/(I guess)buffer.c, and this should be the *only* public interface: no Ex commands, no VimL functions. Tests in test/functional/api.

Note: public interface. Implementation in mark_extended.c.